modules/launchd/default.nix

```diff
@@ -88,7 +88,11 @@ let
         '');
 
         serviceConfig.Label = mkDefault "${cfg.labelPrefix}.${name}";
-        serviceConfig.ProgramArguments = mkIf (cmd != "") [ "/bin/sh" "-c" "exec ${cmd}" ];
+        serviceConfig.ProgramArguments = mkIf (cmd != "") [
+          "/bin/sh"
+          "-c"
+          "/bin/wait4path /nix/store &amp;&amp; exec ${cmd}"
```

Comment:
Is there any special reason, you used `exec` here? I think we could just drop it and it would make extending the cmd string easier, because one could still use basic shell syntax.

Reply:
Using 'exec' was the previous behaviour. I also received a comment (https://github.com/LnL7/nix-darwin/pull/1052#issuecomment-2323053459) suggesting that I add it back. 

I'm now of the view that chaining commands together with any special shell syntax belongs in a script. The shell that is used to execute the program arguments is also not obvious to the user of the abstraction. By using a script one would expect (by nix convention) that you are getting bash, which also has some extended syntax not supported by /bin/sh. 

Totally open to anything else tho